### PR TITLE
[!!!][TASK] Flow 4.0 compatibility

### LINKED ISSUE #49: Move Flow/Neos specific classes to Neos namespace
- [x] Move classes to new namespace
- [x] Adapt cli commands to work with older Flow versions and new ones

### PR DESCRIPTION
**Pull request by Sebobo**
This PR resolves #49.

### CODE REVIEW

File: src/Application/Neos/Flow.php [hunk 1]
```diff
@@ -165,8 +165,36 @@ public function getFlowScriptName()
...
+     * Returns a executable flow command including the context
+     *
+     * @param string $targetPath the path where the command should be executed
+     * @param string $command the actual command for example `cache:flush`
+     * @param array $arguments list of arguments which will be appended to the command
+     * @return string
+     */
+    public function buildCommand($targetPath, $command, array $arguments = [])
+    {
+        return 'cd ' . $targetPath . ' && FLOW_CONTEXT=' . $this->getContext() . ' ./' . $this->getFlowScriptName() . ' ' . $this->getCommandPackageKey() . ':' . $command . ' ' . join(' ', $arguments);
```
Comment (helhum) [inline src/Application/Neos/Flow.php]: all arguments should be properly escaped here

**Comment by Sebobo** [comment]
@helhum you mean doing a whole cleanup of the imports first for all classes?
Then namespace change & compatibility?

**Comment by helhum** [comment]
yes exactly. that would be great

**Comment by langeland** [comment]
I'm not able to install this with composer.. I'm getting:

```
Problem 1
    - Conclusion: remove neos/flow 4.0.0
    - Installation request for neos/flow ~4.0.0 -> satisfiable by neos/flow[4.0.0].
    - typo3/surf dev-master requires neos/utility-files ^3.0 -> satisfiable by neos/utility-files[3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4, 3.2.5, 3.2.6, 3.2.7, 3.2.8, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4, 3.3.5, 3.3.6, 3.3.7].
    - typo3/surf dev-master requires neos/utility-files ^3.0 -> satisfiable by neos/utility-files[3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4, 3.2.5, 3.2.6, 3.2.7, 3.2.8, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4, 3.3.5, 3.3.6, 3.3.7].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.0].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.1].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.2].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.3].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.4].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.5].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.6].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.7].
    - Can only install one of: neos/utility-files[4.0.0, 3.2.8].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.0].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.1].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.2].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.3].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.4].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.5].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.6].
    - Can only install one of: neos/utility-files[4.0.0, 3.3.7].
    - neos/flow 4.0.0 requires neos/utility-files 4.0.0 -> satisfiable by neos/utility-files[4.0.0].
    - Conclusion: remove neos/utility-files 4.0.0
    - Installation request for typo3/surf dev-master#flow-4-compatibility -> satisfiable by typo3/surf[dev-master].
```

**Comment by Sebobo** [comment]
@langeland Ah right this is a problem if you install surf as dependency with the new Flow or Neos versions. I'm currently only using Surf as standalone, so it's not a problem.
For this we might actually need two branches. Because of the dependency on `neos/utility-files`.
Either it's a Flow 3 compatible version or Flow 4. But I'm not sure if we can have it compatible with both then.

Sorry this change takes a while from my side, but I'm currently a bit drowning in other stuff.

**Comment by kdambekalns** [comment]
Updated the README.md a bit to follow the code changes.

**Comment by helhum** [comment]
Once this is in, I would release a new version of Surf

**Comment by ochorocho** [comment]
@Sebobo Hey, seems good to me as long as i do not run `SetFilePermissionsTask` (tested on master)
When running the following task:
```$workflow->addTask('TYPO3\\Surf\\Task\\Neos\\Flow\\SetFilePermissionsTask', 'finalize', $application);```

i get
```
XXXXXX (Buecherei) TYPO3\Surf\Task\Neos\Flow\SetFilePermissionsTask
PHP Warning:  escapeshellarg() expects parameter 1 to be string, array given in /Users/jochen/.composer/vendor/typo3/surf/src/Application/Neos/Flow.php on line 200
```

thanks.

**Pull request merged**